pkg/config/expand.go

```diff
@@ -472,6 +525,48 @@ func (dg *DeploymentGroup) identifySimpleVariable(yamlReference string) (varRefe
 	return ref, nil
 }
 
+func (ref *modReference) validate(depGroups []DeploymentGroup) error {
+	callingModuleGroupIndex := slices.IndexFunc(depGroups, func(d DeploymentGroup) bool { return d.Name == ref.SourceGroupID })
+	if callingModuleGroupIndex == -1 {
+		return fmt.Errorf("%s: %s", errorMessages["groupNotFound"], ref.SourceGroupID)
+	}
+
+	// it's important to lookup where the target module actually exists
+	// and not simply trust the ref.GroupID
+	modToGrp, err := checkModuleAndGroupNames(depGroups)
```

Comment:
I don't really like passing custom objects (`modToGrp`) around but I do understand the objection to calling `checkModuleAndGroupNames` repeatedly which is probably something like O(total-chars-in-module-names). That could change in the future without it being obvious that it would impact this function. So, I'll modify it to pass in `modToGrp`.

If this were Python, I'd throw `@cache` to memoize `checkModuleAndGroupNames`. Golang has an internal [memoize] packagebut it's versioning and relative lack of development doesn't suggest investing in it.

[memoize]: https://pkg.go.dev/golang.org/x/tools/internal/memoize